### PR TITLE
DO NOT MERGE - MULE-11786 - Support API for creating an isolated Application

### PR DESCRIPTION
**Pull request by pablolagreca**

### CODE REVIEW

File: src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+/**
+ * This object matches the mule-policy.json element within a policy. The describer holds information that has being picked up from
+ * the JSON file (and the pom.xml when implemented).
+ *
+ * @since 1.0
+ */
+public class MuleApplicationModel extends AbstractMuleArtifactModel {
+
+  private List<String> configs = new ArrayList<>();
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java]: make final

File: src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                               MuleArtifactLoaderDescriptor bundleDescriptor, List<String> configs) {
+    super(name, minMuleVersion, classLoaderModelLoaderDescriptor, bundleDescriptor);
+    this.configs = configs;
+  }
+
+  /**
+   * @return the application configuration files
+   */
+  public List<String> getConfigs() {
+    return configs;
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java]: return as an unmodifiableList

File: src/main/java/org/mule/runtime/api/deployment/persistence/MuleApplicationModelJsonSerializer.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.api.deployment.persistence;
+
+import org.mule.runtime.api.deployment.meta.MuleApplicationModel;
+
+public class MuleApplicationModelJsonSerializer extends AbstractMuleArtifactModelJsonSerializer<MuleApplicationModel> {
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/deployment/persistence/MuleApplicationModelJsonSerializer.java]: add javadoc

File: src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This object matches the mule-policy.json element within a policy. The describer holds information that has being picked up from
+ * the JSON file (and the pom.xml when implemented).
+ *
+ * @since 1.0
+ */
+public class MuleApplicationModel extends AbstractMuleArtifactModel {
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java]: where is this used? didn't find any references in the related PRs

Comment (pablolagreca) [inline src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java]: ApplicationDescriptorFactory is creating it. That's used internally later in runtime tooling client which will expose the descriptor.

File: src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java (hunk 1 above)
Comment (pablolagreca) [inline src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java]: done

File: src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+  private final List<String> configs;
+  private final String domain;
+  private boolean redeploymentEnabled = true;
```
Comment (elrodro83) [inline src/main/java/org/mule/runtime/api/deployment/meta/MuleApplicationModel.java]: make final, remove init from here, it's already done in the constructor

**Pull request merged**